### DISCUSSION TITLE
Baremetal Rook Cluster expansion

### DISCUSSION THREAD

**Discussion by jjsteel** [rook/rook#7450] [General]
Hello I have a similar question as already asked in #7367 . I've added volumes to my nodes before rook installation, and once rook was successful with the installation process and provisioning I was able to use the pool and assign storage to PVCs.

Further I wanted to know if I add a new volume would it be automatically provisioned and later expand the existing pool.
So I added new volume to one of the nodes but nothing happened as I took the bare-metal approach for rook installation.

After analysing the installation/provisioning  steps I noticed that `job rook-ceph-osd-prepare-XXX` is what I needed to run to provision the new volume to expand the existing pool.

Creating a new job from the existing one to prepare the newly attached volume worked and I saw the newly provisioned OSD part of the existing ceph cluster but it was in a down/unavailable state.

Later I redeployed the rook cluster by first removing all resources under `roon-ceph` namespace and to no surprise the cluster was redeployed with the newly added volume part of if.

 Question is what have I done wrong as I thought the by just executing rook-ceph-osd-prepare job against the target node would have been enough to provision the newly added volume and expand the existing cluster ?

**Comment by jjsteel** [comment]
To answer my own question after further investigation I would need to create new `deployment rook-ceph-osd-%d` to make the prepared volume part of the cluster.  Is this assumption correct ? (Don't have enough time to test this currently.)

**Comment by galexrt** [comment]
You don't need to manually create any jobs or deployments. Depending on if the Rook Ceph operator picks up the new node automatically, the Rook Ceph operator takes care of adding the new nodes. If you have the Rook Discovery daemon active new devices should also be picked up automatically after the configured scan interval.

If the operator didn't react, deleting the Rook Ceph Operator Pod normally gets it going. Make sure to check the Rook Ceph operator logs for any errors and also if a device isn't being picked up correctly check the Rook Ceph OSD Prepare Pods logs for infos.
